rules/framework.bzl

```diff
@@ -63,13 +64,28 @@ def _framework_packaging(ctx, action, inputs, outputs, manifest = None):
     args.add("--action", action)
     args.add_all("--inputs", inputs)
     args.add_all("--outputs", outputs)
-    ctx.actions.run(
-        executable = ctx.executable._framework_packaging,
-        arguments = [args],
-        inputs = action_inputs,
-        outputs = outputs,
-        mnemonic = "PackagingFramework%s" % action.title().replace("_", ""),
-    )
+
+    if action in actions_to_symlink:
+        for input in inputs:
+            input_basename = input.basename
+            output = [x for x in outputs if x.basename == input_basename]
```

Comment:
this is going to lead to `O(n^2)` runtime, would it be possible to make either a list of tuples or a dict of input file -> output file?

Reply:
That makes sense! Will see how I can improve the time complexity here

Reply:
Just curious, what is an example of a basename?

Reply:
Posted a new comment with answers to these 👇